packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+/**
+ * Takes the string value from an input and returns a string
+ * with appropriate date format masking applied concatenated
+ * with the date hint text, MM/DD/YYYY.
+ * @param {String} rawInput
+ * @param {Boolean} valueOnly - Defaults to false. If true,
+ * returns string without additional hint text.
+ * @returns {String}
+ */
+function maskDate(rawInput = '', valueOnly = false): any {
+  const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;
+  const match = RE_DATE.exec(rawInput);
+  let formattedDate = '';
+  if (match) {
+    const [month, day, year] = match.slice(1);
+    formattedDate = [
+      // We treat all non-numeric characters as a delimiter. If they're using a
+      // delimiter after a month or day, we interpret that as the user supplying
+      // a single digit for month or day, which we will automatically pad for them.
+      month && month.padStart(2, '0'),
+      day && day.padStart(2, '0'),
+      year,
+    ]
+      .filter((s) => s)
+      .join('/');
+  }
+  if (valueOnly) {
+    return formattedDate;
+  }
+  const hint = 'MM/DD/YYYY';
+  const hintSub = hint.substring(formattedDate.length);
+  return formattedDate + hintSub;
+}
+
+export const BuiltInMask = {
+  DATE: maskDate,
+};
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  /**
+   * Applies date format masking to the input value entered
+   * and renders to a text field above the input.
+   * Passing `true` to `valueOnly` will return just the
+   * formatted value entered.
+   */
+  labelMask?: (rawInput: string, valueOnly?: boolean) => string;
+}
+
+const LabelMask = ({ children, labelMask }: LabelMaskProps) => {
+  /**
+   * Get the child text field. Called as a method so that
+   * updates to the field cause the mask to re-render
+   * @returns {React.ReactElement} Child TextField
```

Comment:
Because this isn't a function anymore, it doesn't need to be formatted like a JSDOC and include the `@returns` line. It's still a useful comment overall though.